src/aleph/vm/hypervisors/firecracker/microvm.py

```diff
@@ -318,15 +319,23 @@ def enable_rootfs(self, path_on_host: Path) -> Path:
     def enable_file_rootfs(self, path_on_host: Path) -> Path:
         """Make a rootfs available to the VM.
 
-        Creates a symlink to the rootfs file if jailer is in use.
+        If jailer is in use, try to create a hardlink
+        If it is not possible to create a link because the dir are in separate device made a copy.
         """
         if self.use_jailer:
             rootfs_filename = Path(path_on_host).name
             jailer_path_on_host = f"/opt/{rootfs_filename}"
             try:
-                os.link(path_on_host, f"{self.jailer_path}/{jailer_path_on_host}")
+                os.symlink(path_on_host, f"{self.jailer_path}/{jailer_path_on_host}")
```

Comment:
```suggestion
                os.link(path_on_host, f"{self.jailer_path}/{jailer_path_on_host}")
```
So here we need to restore the hardlink also.